tests/Keboola/DbExtractor/RedshiftIncrementalTest.php

```diff
@@ -0,0 +1,352 @@
+<?php
+
+declare(strict_types=1);
+
+namespace Keboola\DbExtractor\Tests;
+
+use Keboola\Csv\CsvFile;
+use Keboola\DbExtractor\Exception\UserException;
+use Keboola\DbExtractorConfig\Exception\UserException as ConfigUserException;
+
+class RedshiftIncrementalTest extends AbstractRedshiftTest
+{
+
+    public function testIncrementalFetchingByTimestamp(): void
+    {
+        $config = $this->getConfigRow();
+        $config['parameters']['incrementalFetchingColumn'] = 'timestamp';
+        $config['parameters']['table']['tableName'] = 'auto_increment_timestamp';
+        $config['parameters']['outputTable'] = 'in.c-main.auto-increment-timestamp';
+        $config['parameters']['columns'] = [];
+        $this->createAutoIncrementAndTimestampTable($config);
+        $app = $this->createApplication($config);
+
+        $result = $app->run();
+        $this->assertEquals('success', $result['status']);
+        $this->assertEquals(
+            [
+                'outputTable' => 'in.c-main.auto-increment-timestamp',
+                'rows' => 2,
+            ],
+            $result['imported']
+        );
+
+        //check that output state contains expected information
+        $this->assertArrayHasKey('state', $result);
+        $this->assertArrayHasKey('lastFetchedRow', $result['state']);
+        $this->assertNotEmpty($result['state']['lastFetchedRow']);
+
+        sleep(2);
+        // the next fetch should be empty
+        $noNewRowsResult = ($this->createApplication($config, $result['state']))->run();
+        $this->assertEquals(1, $noNewRowsResult['imported']['rows']);
+
+        $this->insertRowToTable($config, ['weird-Name' => 'charles']);
+        $this->insertRowToTable($config, ['weird-Name' => 'william']);
+
+        $newResult = ($this->createApplication($config, $result['state']))->run();
+
+        //check that output state contains expected information
+        $this->assertArrayHasKey('state', $newResult);
+        $this->assertArrayHasKey('lastFetchedRow', $newResult['state']);
+        $this->assertGreaterThan(
+            $result['state']['lastFetchedRow'],
+            $newResult['state']['lastFetchedRow']
+        );
+        $this->assertEquals(3, $newResult['imported']['rows']);
+    }
+
+    public function testIncrementalFetchingByDatetime(): void
+    {
+        $config = $this->getConfigRow();
+        $config['parameters']['incrementalFetchingColumn'] = 'datetime';
```

Comment:
I don't see how this test passed,  it should have fallen into `invalid type for incremental fetching, no?` .   I mean, it should be allowed, but it's not in the list https://github.com/keboola/db-extractor-redshift/pull/39/files#diff-30f87cc5e420b0077cb47eb11963d495R12

Reply:
no no... if i create a datetime column, so redshift saved it as a timestamp

![image](https://user-images.githubusercontent.com/12143866/68656556-59b8ac00-0532-11ea-9b6a-fe8572006aff.png)